metadata-ingestion/src/datahub/specific/dashboard.py

```diff
@@ -405,3 +407,119 @@ def remove_custom_property(self, key: str) -> "DashboardPatchBuilder":
         """
         self.custom_properties_patch_helper.remove_property(key)
         return self
+
+    def set_title(self, title: str) -> "DashboardPatchBuilder":
+        assert title, "DashboardInfo title should not be None"
+        self._add_patch(
+            DashboardInfo.ASPECT_NAME,
+            "add",
+            path="/title",
+            value=title,
+        )
+
+        return self
+
+    def set_description(self, description: str) -> "DashboardPatchBuilder":
+        assert description, "DashboardInfo description should not be None"
+        self._add_patch(
+            DashboardInfo.ASPECT_NAME,
+            "add",
+            path="/description",
+            value=description,
+        )
+
+        return self
+
+    def add_custom_properties(
+        self, custom_properties: Optional[Dict[str, str]] = None
+    ) -> "DashboardPatchBuilder":
+
+        if custom_properties:
+            for key, value in custom_properties.items():
+                self.custom_properties_patch_helper.add_property(key, value)
+
+        return self
+
+    def set_external_url(self, external_url: Optional[str]) -> "DashboardPatchBuilder":
+        if external_url:
+            self._add_patch(
+                DashboardInfo.ASPECT_NAME,
+                "add",
+                path="/externalUrl",
+                value=external_url,
+            )
+        return self
+
+    def set_charts(self, charts: Optional[List[str]]) -> "DashboardPatchBuilder":
+        if charts:
+            self._add_patch(
+                DashboardInfo.ASPECT_NAME,
+                "add",
+                path="/charts",
+                value=charts,
+            )
+
+        return self
+
+    def set_datasets(self, datasets: Optional[List[str]]) -> "DashboardPatchBuilder":
+        if datasets:
+            self._add_patch(
+                DashboardInfo.ASPECT_NAME,
+                "add",
+                path="/datasets",
+                value=datasets,
+            )
```

Comment:
Array fields function as maps, this needs to specify the extension at which the value is being added to in the path if it is expected to truly function as a patch, otherwise it will be treated as a full replacement.

Reply:
Could you please give me the reference example

Reply:
Hi @RyanHolstien
I enhanced the code. Please check and confirm if it is ok. I verified this on local instance.